File: typescript-jest/hiker.test.ts
import {part1, part2} from './hiker'
import * as fs from 'fs';

describe('answer', () => {

    describe('Part 1', () => {
        it('sample ', function () {
            const testInput = readInput("input-sample.txt")
            expect(part1(testInput)).toBe(42)
        });

        // Remove .skip
        it.skip('real', function () {
            const testInput = readInput("input.txt")
            expect(part1(testInput)).toBe(1)
        });
    });

    //Remove .skip
    describe.skip('Part 2', () => {

        //Remove .skip
        it.skip('sample ', function () {
            const testInput = readInput("input-sample.txt")
            expect(part2(testInput)).toBe(1)
        });

        //Remove .skip
        it('real', function () {
            const testInput = readInput("input.txt")
            expect(part2(testInput)).toBe(1)
        });
    });

})


function readInput(fileName="input.txt"){
  return fs.readFileSync(fileName, 'utf8').split("\n");
}

File: kotlin/src/HikerTest.kt
    package hiker
    import io.kotlintest.specs.StringSpec
    import io.kotlintest.shouldBe
    import io.kotlintest.shouldNotBe
    import hiker.readInput as readInput

    class HikerTest : StringSpec() {

        init {
            "Part 1 sample" {

                val testInput = readInput("input-sample")
                hiker.part1(testInput) shouldBe 41
            }

            "Part 1 result" {
                val testInput = readInput("input")
                hiker.part1(testInput) shouldBe 42
            }

            "Part 2 sample" {
                val testInput = readInput("input-sample")
                hiker.part2(testInput) shouldBe 42
            }

            "Part 2 result" {
                val testInput = readInput("input")
                hiker.part2(testInput) shouldBe 42
            }


        }
    }